Misc/NEWS.d/next/Library/2024-10-25-10-53-56.gh-issue-125966.eOCYU_.rst

```diff
@@ -0,0 +1,2 @@
+Fix a use-after-free crash in :meth:`asyncio.Future.remove_done_callback`.
```

Comment:
It could be worth noting the change to PyFuture here also

Reply:
Ah right. I forgot about it. I'm not entirely sure that it's the correct way to do it though (hence the separate commit). If people mess up with the callbacks like this, I think we can return garbage (I got -1 as the number of callbacks being removed by the way).